app/javascript/components/action-item/action-item.jsx

```diff
@@ -12,8 +21,111 @@ const ActionItem = ({
   times_moved,
   assignee,
   users,
-  isPrevious
+  isPrevious,
+  author
 }) => {
+  const [actionItemBody, setActionItemBody] = useState(body);
+  const [editMode, setEditMode] = useState(false);
+  const [showDropdown, setShowDropdown] = useState(false);
+  const [actionItemAssigneeId, setActionItemAssigneeId] = useState(
+    assignee?.id
+  );
+  const [destroyActionItem] = useMutation(destroyActionItemMutation);
+  const [updateActionItem] = useMutation(updateActionItemMutation);
+
+  useEffect(() => {
+    if (!editMode) {
+      setActionItemBody(body);
+    }
+  }, [body, editMode]);
+
+  const handleDeleteClick = () => {
+    hideDropdown();
+    destroyActionItem({
+      variables: {
+        id
+      }
+    }).then(({data}) => {
+      if (!data.destroyActionItem.id) {
+        console.log(data.destroyActionItem.errors.fullMessages.join(' '));
+      }
+    });
+  };
+
+  const handleEditClick = () => {
+    editModeToggle();
+    hideDropdown();
+  };
+
+  const editModeToggle = () => {
+    setEditMode(!editMode);
+  };
+
+  const resetTextChanges = () => {
+    setActionItemBody(body);
+  };
+
+  const handleKeyPress = (evt) => {
+    if (navigator.platform.includes('Mac')) {
+      if (evt.key === 'Enter' && evt.metaKey) {
+        editModeToggle();
+        handleItemEdit(id, actionItemBody);
+      }
+    } else if (evt.key === 'Enter' && evt.ctrlKey) {
+      editModeToggle();
+      handleItemEdit(id, actionItemBody);
+    }
+  };
```

Comment:
It could be moved to helpers like this

```js
const handleEnterKeyPress = (evt) => (onEnter) => {
    if (navigator.platform.includes('Mac')) {
      if (evt.key === 'Enter' && evt.metaKey) {
        onEnter();
      }
    } else if (evt.key === 'Enter' && evt.ctrlKey) {
      onEnter();
    }
  };
```